src/Formatter/AssetBaseDirFormatter.php

```diff
@@ -2,32 +2,58 @@
 
 namespace Sensiolabs\GotenbergBundle\Formatter;
 
+use Symfony\Component\Filesystem\Exception\FileNotFoundException;
 use Symfony\Component\Filesystem\Path;
 
 /**
  * @internal
  */
 final class AssetBaseDirFormatter
 {
-    private readonly string $baseDir;
+    /** @var string[] */
+    private readonly array $baseDir;
 
+    /** @var array<string, string> */
+    private array $resolvedPathsCache = [];
+
+    /**
+     * @param string[] $baseDir
+     */
     public function __construct(
         private readonly string $projectDir,
-        string $baseDir,
+        array $baseDir,
     ) {
-        $this->baseDir = rtrim($baseDir, '/\\');
+        $this->baseDir = array_map(static fn ($value) => rtrim($value, '/\\'), $baseDir);
     }
 
     public function resolve(string $path): string
     {
-        if (Path::isAbsolute($path)) {
-            return $path;
+        if (\array_key_exists($path, $this->resolvedPathsCache)) {
+            return $this->resolvedPathsCache[$path];
+        }
+
+        if (Path::isAbsolute($path) || filter_var($path, \FILTER_VALIDATE_URL)) {
+            return $this->resolvedPathsCache[$path] = $path;
         }
 
-        if (Path::isAbsolute($this->baseDir)) {
-            return Path::join($this->baseDir, $path);
+        foreach ($this->baseDir as $baseDir) {
+            if (Path::isAbsolute($baseDir)) {
+                $filename = Path::join($baseDir, $path);
+                if (!file_exists($filename)) {
+                    continue;
+                }
+
+                return $this->resolvedPathsCache[$path] = $filename;
+            }
+
+            $filename = Path::join($this->projectDir, $baseDir, $path);
+            if (!file_exists($filename)) {
+                continue;
+            }
+
+            return $this->resolvedPathsCache[$path] = $filename;
```

Comment:
```suggestion
        foreach ($this->baseDir as $baseDir) {
            if (!Path::isAbsolute($baseDir)) {
                $baseDir = Path::join($this->projectDir, $baseDir);
            }

            $filename = Path::join($baseDir, $path);
            if (!file_exists($filename)) {
                continue;
            }

            return $this->resolvedPathsCache[$path] = $filename;
```

Something like this ?

Reply:
done and thanks :wink: